### PR TITLE
Modifications to work with current keras

### PR DESCRIPTION
**Pull request by elbamos**
Will only work with theano backend until the tensorflow backend supports K.dot().  At that point, will have to add code to test for the running backend, and for a random binomial stream from tensorflow.  

Tested by running examples.py

### CODE REVIEW

**Comment by wuaalb** [comment]
Great, thanks!

Surely K.dot() is supported in TF? Otherwise Dense wouldn't work either..
I'll have a quick review..

Would be nice to have a proper example (e.g. MNIST) for testing.

File: keras_extensions/backend.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from keras import backend as K
+import numpy as np
+from theano.sandbox.rng_mrg import MRG_RandomStreams as RandomStreams
+
+def random_binomial(shape, n=0.0, p=1.0, dtype=K.floatx(), seed=None):
```
Comment (wuaalb) [inline keras_extensions/backend.py]: I'd change this to `n=0, p=0.5`; `n` should be an integer..
Also, I think setting the seed and generating a random number should be separate functions.

File: keras_extensions/backend.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    if seed is None:
+        seed = np.random.randint(10e6)
+    rng = RandomStreams(seed=seed)
```
Comment (wuaalb) [inline keras_extensions/backend.py]: There shouldn't be a new stream each time there's a value generated.

File: keras_extensions/rbm.py [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
-from keras.utils.theano_utils import shared_zeros
+from keras import backend as K 
 from keras.layers.core import Layer, Dense
 
-from theano.sandbox.rng_mrg import MRG_RandomStreams as RandomStreams
+from .backend import random_binomial
+
+import theano
 
 class RBM(Layer):
```
Comment (wuaalb) [inline keras_extensions/rbm.py]: I haven't kept up with Keras much, but this code was based on `keras.core.Dense` and that looks quite a bit different now (e.g. there's a `build`, `output_shape`, ..)..

**Comment by elbamos** [comment]
@wuaalb  --

1 - on K.dot() you're right, that was my goof, K.dot() works on tensorflow to the extent we needed it.  (The support isn't there yet for dot merges but that's another story. 
2.  Ok on changing the defaults for binomials. 
3.  I'm not disagreeing with you about how the random API should look in principle, but I think the most important thing is to conform to what the rest of keras is doing.  See:  https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py   
4.  On build() and output_shape(), that's a longer story and I've been thinking about that.  They would be necessary for the keras functionality of embedding one Model inside another.  Right now there's no facility in keras for designating a layer as having a pre-training mode.  Without that, I agree putting an RBM layer inside another model wouldn't make sense.  

Trying to add pre-training to keras would be a larger project.  I've gone back and forth about whether I want that to be *my* project though.  Right now, since keras doesn't have a working autoencoder class either, there isn't demand yet for a pretraining infrastructure.  The other thing I've been thinking about is adding in functionality for more exotic types of RBMs, with context, etc.  

In terms of the goal of this PR, which is to get keras_extensions working out-of-box with the current version of keras, I think the PR does that.  I'm going to modify the defaults on binomial and re-push.

**Comment by wuaalb** [comment]
Cool, merged.. :)
